### DISCUSSION TITLE
Segmentation Support

### DISCUSSION THREAD

**Discussion by donalSE** [obss/sahi#950] [Q&A] [answered]
Hi,

Just wondering is segmentation fully operational with SAHI at the moment? I have a custom segmentation model, however when I run it, predictions always return with 'mask=None'. 

I've read through all segmentation related topics on the Discussions board and it seems as if this problem isn't new. Also quite sure everything is setup to integrate masks should they be detected. Interested to hear any thoughts or suggestions.

**Code:**
``
class SahiDetection:
    def __init__(self, model_path: str = 'seg-5epoch.pt', config_path: str = 'data.yaml'):
        self.device = 'cuda' if torch.cuda.is_available() else 'cpu'
        print("Using Device: ", self.device)
        self.model = self.load_model(model_path, config_path)

    def load_model(self, model_path, config_path):
        model = AutoDetectionModel.from_pretrained(
            model_type='yolov8',
            model_path=model_path,
            config_path=config_path,
        )
        return model

    def predict(self, frame: np.ndarray):
        height, width, _ = frame.shape
        results = get_sliced_prediction(
            frame,
            self.model,
            slice_height=270,
            slice_width=480,
            overlap_height_ratio=0.2,
            overlap_width_ratio=0.2
        )
        print(results.object_prediction_list)
        return results
      
**Subsequent Print statement:**

bbox: BoundingBox: <(181.3393096923828, 648, 210.01722717285156, 698.9091377258301), w: 28.67791748046875, h: 50.90913772583008>,
    mask: None,
    score: PredictionScore: <value: 0.6274577379226685>,
    category: Category: <id: 4, name: _*hidden*_>>]
``

**Comment by fcakyon** [comment, marked as answer]
Segmentation mask support for yolov8, mmdetection, detectron2, torchvision models is live with the latest release! @donalSE   https://github.com/obss/sahi/discussions/1051

There is also 5x memory and speed improvement when working with masks!

Check the demo notebook on how it works: https://github.com/obss/sahi/blob/main/demo/inference_for_yolov8.ipynb
